include/world_builder/objects/distance_from_surface.h

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright (C) 2018 - 2021 by the authors of the World Builder code.
+
+This file is part of the World Builder.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Lesser General Public License as published
+ by the Free Software Foundation, either version 2 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public License
+ along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+// define a new struct
+#ifndef WORLD_BUILDER_OBJECTSS_DISTANCE_FROM_SURFACE_H
+#define WORLD_BUILDER_OBJECTSS_DISTANCE_FROM_SURFACE_H
+
+namespace WorldBuilder
+{
+  namespace Objects
+  {
+    class PlaneDistances
+    {
+      public:
+
+        /**
+         * Constructor to create an PlaneDistances object from the distance
+         * from and along the surface of plane
+         */
+        PlaneDistances(const double d0, const double d1)
```

Comment:
Could you change the variable names to be more descriptive. Since you can't use the same name, you can name them use the name with an underscrole at the end: `distance_from_surface_`